### ISSUE TITLE
Deserialize doesn't override the whole array 

### ISSUE THREAD

**Issue by erfan71** [jacobdufault/fullserializer#163]
Hi,

suppose that we have an Array of items what we initialize it in the constructor based on some default properties.
so as default, the array size is based on the default array size which is set in the constructor.

but we have changed the array to a smaller one and then we serialize it.
in deserialize time, the Array is still in the default size.
values of indexes from 0 until real saved array size are valid but the rest of it is like what we had set in the constructor.
so In other words, deserializer just overrides part of the array.

Am I doing something wrong? is there any config to set for this?
I can fix this by providing a custom fsObjectProcessor to trim the array OnAfterDeserialize function. is there any better way?

**Comment by GeorgeManeta** [comment]
Found the same issue.
